esmvalcore/dataset.py

```diff
@@ -848,3 +849,21 @@ def _update_timerange(self):
         check.valid_time_selection(timerange)
 
         self.set_facet('timerange', timerange)
+
+    def get_temporary_fixed_file_dir(self) -> Path:
```

Comment:
Could you make this method private (i.e. start the name with an `_`)? Removing things from the public API is difficult and I would like to avoid exposing this functionality. As soon as conversion from an `xarray.DataArray` to an `iris.cube.Cube` works properly (and I know the iris team has plans for this), we can get rid of fix file altogether by loading the file as an uninterpreted `xarray.DataArray`, applying fixes, and then converting to a cube. That will make computations much faster for datasets that require file fixes.

Reply:
I made this public on purpose. `fix_file` (which takes `output_dir` as an argument) is a public function, so I thought it's nice to provide a public method to easily get an output dir. I guess it will take some time until this proper solution via `xarray.DataArray` is available (iris needs to implement it, then we need to implement it). But yeah, I will make it private.